### PR TITLE
Add an actor message passing example

### PR DESCRIPTION
**Pull request by scull7**
## Actix Actor Message Passsing

This is my example repository where I am attempting to understand how to
compose the output of various actors into something that actix_web can use
as a response.

The example shows how to construct a chain of actors (sync) in an asynchronous
way and provides an ergonomic way of doing so.  The important things to not are
the usage of a single error struct `struct SystemError` and the `flatten()`
method to coalesce the future response types.

Hopefully this can be useful and I hope to get some feedback.  Been back using rust for one week, so I'm ready for it all to be considered awful 😄

### CODE REVIEW

**Comment by scull7** [comment]
I understand that this is more about how to compose futures, but using futures within the context of Actix messaging seems to be a different beast because of their slightly different shape.  Also, it seems to me that this type of communication would be common, particularly in the web context.

File: actor-message-passing/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    let sub = params.1;
+
+    state
+        .incr
+        .send(Add { input: add })
+        .join(state.decr.send(Sub { input: sub }))
+        .flatten()
+        .and_then(move |(a, b)| state.sum.send(Sum { a, b }).flatten())
+        .map(|sum| actix_web::HttpResponse::Ok().json(sum))
+        .map_err(actix_web::Error::from)
```
[the remark this answers is not in the document]
Comment (scull7) [reply actor-message-passing/src/main.rs]: Yes, `.from_err` works here, nice catch.

**Comment by scull7** [comment]
With an eye towards future and error mapping ease of use; what existing examples should I look at?

**Comment by DoumanAsh** [comment]
Hmm looking at existing examples, there is one for error handling which overlaps with yours https://github.com/actix/examples/blob/master/error_handling/src/main.rs

Maybe both ideas could be merged

**Comment by scull7** [comment]
I think the example should be updated to use an implementation of `std::convert::From`.  However, I don't know that I would be in favor of adding additional complexity to the example. I believe the example would lose clarity of purpose.

**Comment by DoumanAsh** [comment]
Well, if you look at both examples purpose it is the same I believe: to show graceful error handling?

**Comment by scull7** [comment]
My goal for the example was to demonstrate graceful communication between actors. The execution of this example requires graceful error handling.  Perhaps I didn't achieve my goal.  Is proper goal for an example?  If so, what could I do to improve the effectiveness of the example?

**Comment by DoumanAsh** [comment]
Hmmm I mean that both examples currently try to acheive the same goal: to show how to handle errors, so I think they should be united(improved by your example).

Since both are in domain of error handling, they should be united?

**Comment by scull7** [comment]
I think that your doubt points to the lack of stand-alone utility of this actor communication example.  I'll look to combine them.

**Comment by robjtede** [comment]
Closing because the examples are on v2.0

**Pull request closed without merging**